frontend/src/App.tsx

```diff
@@ -34,12 +37,32 @@ const App = () => {
 
   const dispatch = useAppDispatch();
 
-  useAppRefresh();
+  const latestBlock = useAppSelector(getLatestBlock);
+
+  const { setTimer } = useAppRefresh();
 
   useEffect(() => {
     dispatch(updateBounds(bounds));
   }, [bounds, dispatch]);
 
+  useEffect(() => {
+    if (!latestBlock) {
+      dispatch(fetchLatestBlock());
+    } else {
+      const latestBlockTimeInSeconds =
+        new Date(latestBlock.header.timestamp).getTime() / 1000;
+      const timeNowInSeconds = new Date().getTime() / 1000;
+
+      const blockCreatedTimeAgo = timeNowInSeconds - latestBlockTimeInSeconds;
```

Comment:
Do you think we'd need this code some where else?  Might be useful to slap it in a utility func that could be unit tested.  Thoughts?

Reply:
Yup, good call. I moved it over to a file in the utils folder.